batch_uploads_imageuploader.pl

```diff
@@ -229,13 +229,34 @@ =head2 Methods
        	    print STDERR "Please leave the patient name blank for phantom "
        	                 . "entries\n";
 	        exit $NeuroDB::ExitCodes::PNAME_FILENAME_MISMATCH;
-	}
-	else {
-	    $patientname = 'NULL';
-	}
+        }
+        else {
+            $patientname = 'NULL';
+        }
     }
 
-    ## Populate the mri_upload table with necessary entries and get an upload_id 
+    ## Check the subject information before inserting anything
+
+    if ($phantom eq 'N') {
+        my $python_config = $configOB->getPythonConfigFile();
+
+        my $command = sprintf(
+            "validate_subject_ids.py --profile %s --subject %s",
+            $python_config,
+            $patientname,
+        );
+
+        if ($verbose) {
+            $command .= " --verbose";
+        }
+
+        if (system($command) != 0) {
+            # The error is already printed by the Python script, just exit
+            exit $NeuroDB::ExitCodes::CANDIDATE_MISMATCH;
```

Comment:
This would fail silently. I think the data should still be inserted in the imaging uploader but with a failure status instead.